awscli/autoprompt/output.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import argparse
+import io
+import re
+
+import jmespath
+from botocore.utils import ArgumentGenerator
+
+from awscli.formatter import get_formatter
+from awscli.autocomplete.local.fetcher import CliDriverFetcher
+
+
+class OutputGetter:
+    def __init__(self, driver):
+        self._cli_driver_fetcher = CliDriverFetcher(driver)
+        self._output_formats = driver.arg_table['output'].choices
+        self._session = driver.session
+        self._current_expression = None
+
+    def get_output(self, parsed):
+        operation_model = self._cli_driver_fetcher.get_operation_model(
+            parsed.lineage, parsed.current_command)
+        if operation_model:
+            output_shape = getattr(operation_model, 'output_shape', None)
+            if self._shape_has_members(output_shape):
+                operation = ''.join(
+                    [part.capitalize()
+                     for part in parsed.current_command.split('-')]
+                )
+                output = self._get_output(parsed)
+                query, error_message = self._get_query(parsed)
+                if error_message is not None:
+                    return error_message
+                return self._get_display(operation, output_shape,
+                                         output, query)
+        return 'No output'
+
+    def _shape_has_members(self, shape):
+        return shape and (getattr(shape, 'members', False) or
+                          getattr(shape, 'member', False))
+
+    def _get_output(self, parsed):
+        if parsed.current_param == 'output':
+            output = parsed.current_fragment
+        else:
+            output = parsed.global_params.get('output')
+        if output not in self._output_formats:
+            output = self._session.get_config_variable('output')
+        return output
+
+    def _get_query(self, parsed):
+        error_message = None
+        if parsed.current_param == 'query':
+            query = parsed.current_fragment
+        else:
+            query = parsed.global_params.get('query')
+        if query:
+            query = re.sub(r'([\{\[])\d+?([\}\]])', '\g<1>0\g<2>', query)
+            query = re.sub(r'([\{\[])[^\]^\}]*$', '', query).strip('.')
+        try:
+            self._current_expression = jmespath.compile(query)
+        except jmespath.exceptions.ParseError as e:
+            error_message = "Bad query %s: %s" % (query, str(e))
```

Comment:
For the error message, Let's make it instead:
```python
error_message = "Bad value for --query: %s\n\n%s" % (query, str(e))
```
I think the main points is that we should use `--` to indicate it is the `--query` parameter and it will add newlines to the error to help make it a little more readable.